Lib/dataclasses.py

```diff
@@ -0,0 +1,776 @@
+    #  override earlier field definitions in base classes.
+    for b in cls.__mro__[-1:0:-1]:
+        # Only process classes that have been processed by our
+        #  decorator.  That is, they have a _MARKER attribute.
+        base_fields = getattr(b, _MARKER, None)
+        if base_fields:
+            for f in base_fields.values():
+                fields[f.name] = f
+
+    # Now find fields in our class.  While doing so, validate some
+    #  things, and set the default values (as class attributes)
+    #  where we can.
+    for f in _find_fields(cls):
+        fields[f.name] = f
+
+        # If the class attribute (which is the default value for
+        #  this field) exists and is of type 'Field', replace it
+        #  with the real default.  This is so that normal class
+        #  introspection sees a real default value, not a Field.
+        if isinstance(getattr(cls, f.name, None), Field):
+            if f.default is _MISSING:
+                # If there's no default, delete the class attribute.
+                #  This happens if we specify field(repr=False), for
+                #  example (that is, we specified a field object, but
+                #  no default value).  Also if we're using a default
+                #  factory.  The class attribute should not be set at
+                #  all in the post-processed class.
+                delattr(cls, f.name)
+            else:
+                setattr(cls, f.name, f.default)
+
+    # Remember all of the fields on our class (including bases).  This
+    #  marks this class as being a dataclass.
+    setattr(cls, _MARKER, fields)
+
+    # We also need to check if a parent class is frozen: frozen has to
+    #  be inherited down.
+    is_frozen = frozen or cls.__setattr__ is _frozen_setattr
+
+    # If we're generating ordering methods, we must be generating
+    #  the eq methods.
+    if order and not eq:
+        raise ValueError('eq must be true if order is true')
+
+    if init:
+        # Does this class have a post-init function?
+        has_post_init = hasattr(cls, _POST_INIT_NAME)
+
+        # Include InitVars and regular fields (so, not ClassVars).
+        _set_attribute(cls, '__init__',
+                       _init_fn(list(filter(lambda f: f._field_type
+                                              in (_FIELD, _FIELD_INITVAR),
+                                            fields.values())),
+                                is_frozen,
+                                has_post_init,
+                                # The name to use for the "self" param
+                                #  in __init__.  Use "self" if possible.
+                                '__dataclass_self__' if 'self' in fields
+                                    else 'self',
+                                ))
+
+    # Get the fields as a list, and include only real fields.  This is
+    #  used in all of the following methods.
+    field_list = list(filter(lambda f: f._field_type is _FIELD,
```

Comment:
Wouldn't the code look clearer if use a list comprehension?

Reply:
Probably so.